bundles/org.eclipse.equinox.p2.touchpoint.natives/src/org/eclipse/equinox/internal/p2/touchpoint/natives/SimpleBackupStore.java

```diff
@@ -271,9 +272,11 @@ public boolean backupDirectory(File file) throws IOException {
 			throw new IllegalArgumentException(NLS.bind(Messages.BackupStore_not_a_directory, file.getAbsolutePath()));
 		}
 
-		if (Files.list(path).count() > 0) {
-			throw new IllegalArgumentException(
-					NLS.bind(Messages.BackupStore_directory_not_empty, file.getAbsolutePath()));
+		try (Stream<Path> s = Files.list(path)) {
+			if (s.count() > 0) {
```

Comment:
I think counting all items is not necessary `s.anyMatch(()->true)` should be sufficent

Reply:
Note that you are asking Simon to improve code that was reviewed and approved previously (last year) which is when this suggestion should have been made.  This is all fine and good, I just point that out because if we require contributors not only to fix a bug but also improve the existing style, we may turn them away.

@drsgoodall Of course the suggestion from @mickaelistria is a very good one.  Also, you will need to change the version in
/org.eclipse.equinox.p2.touchpoint.natives/META-INF/MANIFEST.MF from 1.4.300.qualifier to 1.4.400.qualifier for the build to stop complaining so it should be easy to do both things... Thanks so much for contributing!!

Reply:
@merks as the issue is with unclosed file handles I assume that @drsgoodall is seeing issues with files being locked for this or other processes.
So I think (even though this might fix the intermediate issue as well in the current state) its worth to lock not more files as necessary and exit as early as possible (and thus closing the opened file handle).

I also think the suggestion is not that complex to argue it can't be handled as part of this change?

Reply:
@laeubi I said it was a good suggestion!  I see that @drsgoodall considered it as well.   Thanks!!